rllib/algorithms/algorithm.py

```diff
@@ -2977,6 +2977,16 @@ def log_result(self, result: ResultDict) -> None:
 
     @override(Trainable)
     def cleanup(self) -> None:
+        # Stop all Learners.
+        if hasattr(self, "learner_group") and self.learner_group is not None:
+            self.learner_group.shutdown()
```

Comment:
This does sometimes not work. At least in my workspace running on GPU a `algo.stop()` did not pull down the  `BackendExecutor` workers.